extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1372,12 +1383,14 @@ export class AmpStoryPage extends AMP.BaseElement {
 
     this.mediaPoolPromise_.then(mediaPool => {
       if (visible) {
-        this.registerMedia_(mediaPool, videoEl).then(() => {
-          this.playMedia_(mediaPool, videoEl);
-          if (!this.storeService_.get(StateProperty.MUTED_STATE)) {
-            this.unmuteAllMedia();
-          }
-        });
+        this.registerMedia_(mediaPool, videoEl)
+          .then(() => this.preloadMedia_(mediaPool, videoEl))
```

Comment:
Should this be done in `playMedia_()` instead?

Reply:
Having this in two steps (preload / play) is convenient for two main reasons:
- it keeps `playMedia_` lighter and optimizes play/pause interactions, like hold to pause
- it allows us to start listening for media events (waiting/playing/...) between the preload and the play actions, to avoid extra events sent when updating the sources

Pretty sure we can find workarounds if you feel strongly though. The first point can be fixed by making `preloadMedia_` a noop when needed, and the second one by handling these events differently

Reply:
Sure, just curious if we should enforce the invariant in `playMedia_()` instead of relying on callers to know what to do.